File: 0x00-newComers/Sheet4/12-stringFunction.py
lnth = int(input())
testCases = int(input())
word = input()
if len(word) != lnth:
  raise ValueError(f"The length of the word should not be {lnth} characters!")

while testCases :
  stmt = input()
  parts = stmt.split(" ")
  if parts[0] == 'front':
    print()
  if parts[0] == 'back':
    pass
  if parts[0] == 'sort':
    pass
  if parts[0] == 'reverse':
    pass
  if parts[0] == '':
    pass
  if (parts[0] == 'push_back'):
    word += parts[1]
  testCases -= 1

print(word)

File: Newcomers-Training/Sheet4/12-stringFunction.cpp

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

int main()
{
  int len, testCases;
  cin >> len >> testCases;
  int start, end;
  string s;
  cin >> s;
  if (s.length() != len)
  {
    cout << "String size must equal " << len;
    return 0;
  }
  while (testCases--)
  {
    string ops;
    cin >> ops;

    if (ops == "pop_back")
      s.pop_back();
    else if (ops == "front")
      cout << s.front() << endl;
    else if (ops == "back")
      cout << s.back() << endl;
    else if (ops == "print")
    {
      int index;
      cin >> index;
      cout << s[index - 1] << endl;
    }
    if (ops == "sort")
    {
      cin >> start >> end;
      if (start > end)
        swap(start, end);
      sort(s.begin() + start - 1, s.begin() + end);
    }
    if (ops == "reverse")
    {
      cin >> start >> end;
      if (start > end)
        swap(start, end);
      reverse(s.begin() + start - 1, s.begin() + end);
    }

    if (ops == "substr")
    {
      cin >> start >> end;
      if (start > end)
        swap(start, end);
      cout << s.substr(start - 1, end - start + 1) << endl;
    }

    if (ops == "push_back")
    {
      char c;
      cin >> c;
      s.push_back(c);
    }
  }
  return 0;
}